orttraining/orttraining/core/graph/gradient_builder.cc

```diff
@@ -1059,5 +1059,25 @@ IMPLEMENT_GRADIENT_BUILDER(GetRecvGradient) {
               SrcNodeAttributes())};
 }
 
+IMPLEMENT_GRADIENT_BUILDER(GetExpandGradient) {
+  ArgDef a = I(0), y = O(0);
+  std::vector<Dimension> a_shape = GetShape(a);
+  std::vector<Dimension> y_shape = GetShape(y);
```

Comment:
We should error out if a_shape or y_shape is not populated. 

I checked GetShape() and ComputeBroadcastBackwardAxes(), neither of them checks this.